### ISSUE TITLE
v2 template RepoUrlPicker dropdown not populated with options when frontend is built for production

### ISSUE THREAD

**Issue by ConnorDY** [backstage/backstage#5836]
We're using a built and dockerized version of the Backstage frontend and noticed that one thing wasn't working with the prod version of the frontend. There is a `RepoUrlPicker` dropdown as part of a v2 template that isn't showing any options. We can see the option under the `GET` `/api/scaffolder/v2/templates/default/template/template-name/parameter-schema` request but the dropdown isn't populated with it. It shows up and works as expected when running in dev mode (via `yarn start`).

## Expected Behavior

We'd expect the dropdown to behave the same way as it does in dev mode and be populated with the options from the template configuration.

## Current Behavior

The dropdown doesn't show the options from the template configuration.

## Possible Solution

No idea.

## Steps to Reproduce

1. Dockerize your frontend and backend separately following the instructions from here:
https://backstage.io/docs/deployment/docker
2. Create a v2 template with something like:
```yaml
apiVersion: backstage.io/v1beta2
kind: Template
metadata:
  name: backstage-java-api
  title: Backstage Java API
  description: Cookiecutter template for a sample Java API scaffold.
  tags:
    - java
    - cookiecutter
  spec:
    type: service
    parameters:
      - title: Choose a location
        required:
          - repoUrl
        properties:
          repoUrl:
            title: Repository Location
            type: string
            ui:field: RepoUrlPicker
            ui:options:
              allowedHosts:
                - github.com
```
3. Try to use the template from dockerized/prod Backstage instance and notice that the dropdown isn't populated with `github.com`

## Context

- Separate frontend and backend using Docker
  - Frontend setup very similar to this: [Frontend with Nginx](https://github.com/backstage/backstage/tree/master/contrib/docker/frontend-with-nginx)
  - Backend setup very similar to this: [Multi-stage Build](https://backstage.io/docs/deployment/docker#multi-stage-build)

## Your Environment

- NodeJS Version (v12):
- Operating System and Version (e.g. Ubuntu 14.04):
- Browser Information:

NodeJS Version: v14
Operating System (backend): `node:14-buster-slim` Docker image
Operating System (frontend): MacOS Catalina 10.15.7
Browser Information: Chrome (90.0.4430.212)

**Comment by ConnorDY** [comment]
> Would you be able to supply your `app-config.yaml` to this issue? It could be related to your `intergrations:` section but it would be easier for us to help out with that information. 

We're using environment variables to set the configuration for the frontend via [inject-config.sh](https://github.com/backstage/backstage/blob/2b5b5b54fc9f925ba36cde0a12f3140cd57b163e/contrib/docker/frontend-with-nginx/docker/inject-config.sh).

The frontend `app-config.yaml` we use for building looks like this (but again, all of this is overriden via environment variables):
```yaml
# this is a config file used for building the frontend docker container
# these configuration values can be overridden at runtime via environment variables
app:
  title: __APP_INJECTED_RUNTIME_TITLE__
  baseUrl: http://localhost

backend:
  baseUrl: http://localhost:7000

techdocs:
  requestUrl: http://localhost:7000/api/techdocs
  storageUrl: http://localhost:7000/api/techdocs/static/docs
  builder: 'local'
```

Our integrations are only set in the backend `app-config.yaml`. Do they also need to be set for the frontend?

**Comment by benjdlambert** [comment]
@ConnorDY yes, you'll need them in your frontend config too as some of the values are passed to the frontend. By some I mean most of it but without the `token` or anything secure.

**Comment by ConnorDY** [comment]
> @ConnorDY yes, you'll need them in your frontend config too as some of the values are passed to the frontend. By some I mean most of it but without the `token` or anything secure.

Is this something y'all could document better?

**Comment by Rugvip** [comment]
Closed by the merge of #5873, let us know if you have any other concerns!
